src/lib/account/cosmosSdk.ts

```diff
@@ -18,6 +20,12 @@ export const deriveCosmosSdkAccountIdsAndMetadata: DeriveAccountIdsAndMetadata =
       if (chainReference === CHAIN_REFERENCE.ThorchainMainnet) {
         if (!supportsThorchain(wallet)) continue
       }
+      if (
+        wallet instanceof MetaMaskShapeShiftMultiChainHDWallet &&
+        !canAddMetaMaskAccount({ accountNumber, chainId, wallet, isSnapInstalled })
+      ) {
+        continue
+      }
```

Comment:
and much better on this and the EVM variant for a more hollistic version, nice one 🧠